solutions/devsprint-eduardo-bocato-2/FinanceAppTests/Mock/FinanceServiceMock.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  FinanceServiceMock.swift
+//  FinanceAppTests
+//
+//  Created by Murillo R. Araújo on 03/05/22.
+//
+
+import Foundation
+@testable import FinanceApp
+
+enum APIError: Error {
+    case noData
+    case parseError
+}
+
+class FinanceServiceMock: FinanceServiceProtocol {
+    var errorAPI: APIError? = nil
+
+    func fetchHomeData(_ completion: @escaping (HomeData?) -> Void) {
+        switch errorAPI {
+            case .parseError:
+                completion(nil)
+            case .noData:
+                completion(nil)
+        default:
+            let homeData = HomeData(balance: 100, savings: 500, spending: 100, activity: [])
+            completion(homeData)
+        }
+    }
+}
```

Comment:
**[CHANGE REQUEST]**
Please use a stub and configure the scenarios in your test functions.
```suggestion
final class  FinanceServiceStub: FinanceServiceProtocol {
    var homeDataToBeReturned: HomeData? = nil
    func fetchHomeData(_ completion: @escaping (HomeData?) -> Void) {
        completion(homeDataToBeReturned)
    }
}
```

Reply:
Done